planetiler-core/src/main/java/com/onthegomap/planetiler/reader/SimpleReader.java

```diff
@@ -125,8 +133,25 @@ private FeatureRenderer newFeatureRenderer(FeatureGroup writer, PlanetilerConfig
   }
 
   /** Returns the number of features to be read from this source to use for displaying progress. */
-  public abstract long getCount();
+  public long getCount() {
+    long numFeatures = 0;
+    for (var path : sourcePaths) {
+      numFeatures += getCountForPath(path);
+    }
+    return numFeatures;
+  }
+
+  /** Returns the number of features to be read from a single path to use for displaying progress. */
+  public abstract long getCountForPath(Path path);
 
   /** Returns a source that initiates a {@link WorkerPipeline} with elements from this data provider. */
-  public abstract WorkerPipeline.SourceStep<? extends SourceFeature> read();
+  public abstract void readPath(Path path, Consumer<F> next) throws Exception;
```

Comment:
Sonar is complaining but I think this is fine since a bunch of different kinds of exceptions can be thrown and we want them all to terminate the whole thing - you can add `@SuppressWarnings("java:S112")` so sonar stops bugging us